consensus/types/src/subnet_id.rs

```diff
@@ -77,48 +76,16 @@ impl SubnetId {
     /// Computes the set of subnets the node should be subscribed to during the current epoch,
     /// along with the first epoch in which these subscriptions are no longer valid.
     #[allow(clippy::arithmetic_side_effects)]
-    pub fn compute_subnets_for_epoch<E: EthSpec>(
+    pub fn compute_attestation_subnets(
         raw_node_id: [u8; 32],
-        epoch: Epoch,
         spec: &ChainSpec,
-    ) -> Result<(impl Iterator<Item = SubnetId>, Epoch), &'static str> {
-        // simplify variable naming
-        let subscription_duration = spec.epochs_per_subnet_subscription;
+    ) -> impl Iterator<Item = SubnetId> {
+        // The bits of the node-id we are using to define the subnets.
         let prefix_bits = spec.attestation_subnet_prefix_bits as u64;
-        let shuffling_prefix_bits = spec.attestation_subnet_shuffling_prefix_bits as u64;
-        let node_id = U256::from_be_slice(&raw_node_id);
 
+        let node_id = U256::from_be_slice(&raw_node_id);
         // calculate the prefixes used to compute the subnet and shuffling
         let node_id_prefix = (node_id >> (256 - prefix_bits)).as_le_slice().get_u64_le();
```

Comment:
Maybe worth creating a constant for `NODE_ID_BITS = 256` for readability

Reply:
Done